clueless/views.py

```diff
@@ -1,16 +1,97 @@
+from django.contrib.auth import authenticate, login as auth_login, logout as auth_logout
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.models import User
+from django.core.exceptions import ObjectDoesNotExist
 from django.http import HttpResponse
-from django.template import loader
-from .models import *
-from clueless import models
+from django.shortcuts import redirect
+from django.template import Context, loader
+from clueless.models import Accusation, Character, Game, Player, Room, STATUS_CHOICES, Suggestion, Weapon, WhoWhatWhere
+import logging
 
-# HttpResponse functions below here 
+# Get an instance of a logger
+logger = logging.getLogger(__name__)
+
+# HttpResponse functions below here
 
 def index(request):
 	#return HttpResponse("Welcome to the world of Clue-Less!")
 	template = loader.get_template('clueless/index.html')
 	context = {}
 	return HttpResponse(template.render(context, request))
 
+def login(request):
+	"""
+	Logs in a user
+	:param request:
+	:return:
+	"""
+
+	#will be set to true if there is an issue logging in
+	loginError = False
+
+	if(request.method == "POST"):
+		#get the username and password, and attempt to authenticate
+		username = request.POST.get('username')
+		password = request.POST.get('password')
+
+		user = authenticate(username=username, password=password)
+		#if the user authenticated, go to the next get param if set, otherwise go to homepage
+		if user is not None:
+			auth_login(request, user)
+			redirectNext = request.GET.get('next')
+			if redirectNext is not None:
+				redirect(redirectNext)
+			else:
+				return redirect('index')
+		else: #user didn't login right, give them a login error
+			loginError = True
+
+	#either the authentication failed, or they are coming to the login page for the first time.  Render the page
+	template = loader.get_template('clueless/login.html')
+	context = Context({'loginError':loginError})
+	return HttpResponse(template.render(context, request))
+
```

Comment:
maybe my lack of database default set ups? i'll see if i still get it after fixing the python issues